torch/csrc/jit/codegen/cuda/lower2device.cpp

```diff
@@ -373,7 +373,7 @@ void GPULower::replaceSizes() {
     }
 
     TensorDomain* old_domain = tv->domain();
-    TensorDomain* new_domain = TransformReplay::fullReplay(
+    TensorDomain* new_domain = TransformReplay::fullReplay( // why do we need to reply here?
```

Comment:
We modified only the root domain, we need to replay the transformations on the new root domain. Can discuss in more detail by voice or slack.

Reply:
thanks - I'm still wrapping my head around this and I may ask you more about it

Reply:
Of course, Naoya and I just went through this recently, it's not an easy concept to grasp clearly, and I/we should write some documentation on it.